Content.Client/Light/EntitySystems/LightBehaviorSystem.cs

```diff
@@ -33,23 +38,145 @@ private void OnBehaviorAnimationCompleted(EntityUid uid, LightBehaviourComponent
     {
         // TODO: Do NOT ensure component here. And use eventbus events instead...
-        EnsureComp<AnimationPlayerComponent>(uid);
+        EnsureComp<AnimationPlayerComponent>(entity);
 
-        foreach (var container in component.Animations)
+        foreach (var container in entity.Comp.Animations)
         {
-            container.LightBehaviour.Initialize(uid, _random, EntityManager);
+            container.LightBehaviour.Initialize(entity, _random, EntityManager);
         }
 
         // we need to initialize all behaviours before starting any
-        foreach (var container in component.Animations)
+        foreach (var container in entity.Comp.Animations)
         {
             if (container.LightBehaviour.Enabled)
             {
-                component.StartLightBehaviour(container.LightBehaviour.ID);
+                StartLightBehaviour(entity, container.LightBehaviour.ID);
+            }
+        }
+    }
+
+    /// <summary>
+    /// If we disable all the light behaviours we want to be able to revert the light to its original state.
+    /// </summary>
+    private void CopyLightSettings(Entity<LightBehaviourComponent> entity, string property)
+    {
+        if (EntityManager.TryGetComponent(entity, out PointLightComponent? light))
+        {
+            var propertyValue = AnimationHelper.GetAnimatableProperty(light, property);
+            if (propertyValue != null)
+            {
+                entity.Comp.OriginalPropertyValues.Add(property, propertyValue);
+            }
+        }
+        else
+        {
+            Log.Warning($"{EntityManager.GetComponent<MetaDataComponent>(entity).EntityName} has a {nameof(LightBehaviourComponent)} but it has no {nameof(PointLightComponent)}! Check the prototype!");
+        }
+    }
+    /// <summary>
+    /// Start animating a light behaviour with the specified ID. If the specified ID is empty, it will start animating all light behaviour entries.
+    /// If specified light behaviours are already animating, calling this does nothing.
+    /// Multiple light behaviours can have the same ID.
+    /// </summary>
+    public void StartLightBehaviour(Entity<LightBehaviourComponent> entity, string id = "")
+    {
+        if (!EntityManager.TryGetComponent(entity, out AnimationPlayerComponent? animation))
+        {
+            return;
+        }
+        foreach (var container in entity.Comp.Animations)
+        {
+            if (container.LightBehaviour.ID == id || id == string.Empty)
+            {
+                if (!_player.HasRunningAnimation(entity, animation, LightBehaviourComponent.KeyPrefix + container.Key))
+                {
+                    CopyLightSettings(entity, container.LightBehaviour.Property);
+                    container.LightBehaviour.UpdatePlaybackValues(container.Animation);
+                    _player.Play(entity, container.Animation, LightBehaviourComponent.KeyPrefix + container.Key);
+                }
+            }
+        }
```

Comment:
```suggestion
        if (!EntityManager.TryGetComponent(entity, out AnimationPlayerComponent? animation))
            return;

        foreach (var container in entity.Comp.Animations)
        {
            if (container.LightBehaviour.ID != id || id != string.Empty
                || _player.HasRunningAnimation(entity, animation, LightBehaviourComponent.KeyPrefix + container.Key))
                continue;
                
            CopyLightSettings(entity, container.LightBehaviour.Property);
            container.LightBehaviour.UpdatePlaybackValues(container.Animation);
            _player.Play(entity, container.Animation, LightBehaviourComponent.KeyPrefix + container.Key);
        }
```
I hate that wizden codes this crap like yanderedev. So many unecessary indentations and curly braces, no guard clauses.